lib/Conversion/TTIRToTTIRDecomposition/TTIRToTTIRDecomposition.cpp

```diff
@@ -1034,6 +1038,99 @@ struct ArangeForceLastDimensionPattern
   }
 };
 
+// TTNN assumes that the input tensor is in NHWC format (channel last). If the
+// input tensor is in NCHW format (channel first), then IR should be
+// transformed from this:
+//   inputs: %arg0: tensor<NxCxHxW>, %dst: tensor<NxCxH'xW'>
+//   %1 = "ttir.upsample"(%arg0, %dst) {channel_last = false}:
+//          (tensor<NxCxHxW>, tensor<NxCxH'xW'>) -> tensor<NxCxH'xW'>
+
+// Into this:
+//   %0 = tensor.empty() : tensor<NxHxWxC>
+//   %1 = "ttir.permute"(%arg0, %0) {permutation = array<i64: 0, 2, 3, 1>}:
+//          (tensor<NxCxHxW>, tensor<NxHxWxC>) -> tensor<NxHxWxC>
+
+//   %2 = tensor.empty() : tensor<NxH'xW'xC>
+//   %3 = "ttir.upsample"(%1, %2) {channel_last = true}:
+//          (tensor<NxHxWxC>, tensor<NxH'xW'xC>) -> tensor<NxH'xW'xC>
+
+//   %4 = "ttir.permute"(%3, %dst) {permutation = array<i64: 0, 3, 1, 2>}:
+//          (tensor<NxH'xW'xC>, tensor<NxCxH'xW'>) -> tensor<NxCxH'xW'>
+class UpsampleToUpsampleConversionPattern
```

Comment:
I would call this something like `CannonicalizeChannelLastUpsample`. Just so you get a quick understanding of what it’s doing.